include/seqan/align/dp_formula.h

```diff
@@ -104,6 +104,25 @@ typedef Tag<RecursionDirectionZero_> RecursionDirectionZero;
 template <typename TCellTuple>
 using ExtractedScoreValueType_ = std::decay_t<decltype(_scoreOfCell(std::get<0>(std::declval<TCellTuple>())))>;
 
+// A transformation trait to select the correct _maxScore overload for checking against 0 in local alignment.
+template <typename TTraceConfig>
+struct TraceConfigForLocalAlignment_
+{
+private:
+    // If not CompleteTrace then use same type.
+    template <typename TType>
+    static auto selectType(TType) -> TType;
+
+    // In last check of local alignment always invoke SingleTrace _maxScore overload.
```

Comment:
I'm not completely understand this.

You change the type `TracebackOn<TracebackConfig_<CompleteTrace, TGapsPlacement>>` if it is a local alignment to `TracebackOn<TracebackConfig_<SingleTrace, TGapsPlacement>>`.

So `CompleteTrace` would result in garbage for LocalAlignments, but `SingleTrace` does the right thing.

I don't comprehend your comment:
> In last check of local alignment always invoke SingleTrace _maxScore overload.

because it suggests that this `SingleTrace` thing will only be invoked in the last check and otherwise `CompleteTrace` will be used. But, it will always be used `SingleTrace`.

I'm also not completely sure what `SingleTrace` and `CompleteTrace` mean. Has `SingleTrace` somehting to do with SIMD? Or has `CompleteTrace` something to do with ALL possible alignment trace paths?

Reply:
Yeah, let's talk about it in the office.